### PR TITLE
Identity Resolution documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Identity resolution documentation

## Type of documentation
- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

> [ID resolution](https://www.notion.so/rudderstacks/Identity-Resolution-doc-6a37dd4e8b454830b0beb22faecf5d43)

### CODE REVIEW

**Review by achettyiitr — APPROVED** [review]
LGTM

File: identity-resolution.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+### Use case 
+
+This use case gives a quick walkthrough of a user's product journey on an e-Commerce app integrated with RudderStack. You will see how RudderStack can be used to track and merge different user identifiers to build a unified customer profile.
+
+1. The user **anonymously** searches for a particular product on their mobile app - in this case, an iPhone. Here, RudderStack identifies the user with `anonymous_id_1` and the device with `device_id_1`.
+2. After browsing through the app, the user adds a product to their cart and **logs in** using their email. Here, RudderStack generates an `identify` event and associates `anonymous_id_1` with the user's email ID `email` (user's unique identifier). Note that RudderStack also generates a `track` call with the details around the browsing activity and updates to the cart.
+
+Then, the user does not proceed with the purchase and drops off, for some reason.
```
Comment (ishachopra0401) [inline identity-resolution.mdx:41]:
```suggestion
3. Suppose, the user does not proceed with the purchase and drops off, for some reason.
```

Can be added as a step I believe. I get it that there is no useful info to add as a step but seems like a lingering sentence as of now. Or maybe merge with the above or below step..
Thoughts?

File: identity-resolution.mdx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+3. Later, the same user fires up their laptop and browses through the website of the e-Commerce brand **anonymously**. Again, RudderStack generates an `anonymous_id_2` corresponding to the user and a device ID `device_id_2`.
```
Comment (ishachopra0401) [inline identity-resolution.mdx:43]:
```suggestion
3. Later, the same user uses a laptop and browses through the e-commerce website **anonymously**. Again, RudderStack generates an `anonymous_id_2` corresponding to the user and a device ID `device_id_2`.
```

File: identity-resolution.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+4. The user then **logs in** with their email and password. RudderStack then generates an `identify` call associating the same `email` with `anonymous_id_2`. The user then proceeds to complete the purchase. A `track` call is then generated with the details around the purchase.
+
+The above workflow is visually represented in the following image:
+
+<img src="./assets/identity-resolution/identity-resolution-workflow.png" alt="Identity Resolution workfow" />
+
+RudderStack has a `merge` API that you can leverage to connect and merge these different user identities in the warehouse and associate them to a single customer profile:
```
Comment (ishachopra0401) [inline identity-resolution.mdx:50]: We can maybe end the use case with the image and merge the below information in the "Merging identifiers with RudderStack" heading. As we are talking about these options below already. Here we can say something like "To merge these different user identities in the warehouse and associate them to a single customer profile, refer to the below section"

Thoughts?

File: identity-resolution.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+</div>
+
+## Merging identifiers with RudderStack
+
+RudderStack gives you two options to merge user identifiers and build an identity graph:
+
+- Using the Merge API
+- Leveraging user transformations
+
+### `merge` API
```
Comment (ishachopra0401) [inline identity-resolution.mdx:66]:
```suggestion
### Using the`merge` API
```

File: identity-resolution.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+You can also leverage RudderStack's <a href="#user-transformations">Transformations</a> feature to merge different identifiers before sending the data to the warehouse.
+</div>
+
+## Merging identifiers with RudderStack
+
+RudderStack gives you two options to merge user identifiers and build an identity graph:
+
+- Using the Merge API
+- Leveraging user transformations
+
```
Comment (ishachopra0401) [inline identity-resolution.mdx:65]: Can remove these pointers as they will appear on the right under "Contents"

File: identity-resolution.mdx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+### `merge` API
+
+As discussed in the use case above, RudderStack provides a `merge` API to connect and merge these different user identities in the warehouse.
```
Comment (ishachopra0401) [inline identity-resolution.mdx:68]:
```suggestion
As discussed in the use case above, RudderStack provides the `merge` API to connect and merge these different user identities in the warehouse.
```

File: identity-resolution.mdx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            "value": "merge_property_value_2"
+        }
+    ]
+}
+
+Here, `"type": "merge_property_type_*"` corresponds to a unique user identifier like email address, phone number, device ID, anonymous ID, etc. that can be merged and connected to a given user profile. 
+
+### User transformations
+
+As an alternative to the `merge` API, you can also use [custom transformations](https://www.rudderstack.com/docs/transformations/) to merge user identifiers in your event data before sending it to the warehouse destination.
```
Comment (ishachopra0401) [inline identity-resolution.mdx:90]:
```suggestion
Alternatively, you can use [custom transformations](https://www.rudderstack.com/docs/transformations/) to merge user identifiers in your event data before sending it to the warehouse destination.
```

File: identity-resolution.mdx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+<img src="./assets/identity-resolution/identity-mappings-table.png" alt="Identity Mappings table in warehouse" />
+
+On the other hand, the `RUDDER_IDENTITY_MERGE_RULES` table contains all the edges that connect two different identifiers (nodes).
+
+<img src="./assets/identity-resolution/identity-merge-rules.png" alt="Identity merge rules table in the warehouse" />
+
+## Enriching the identity graph
+
+As the user logs into more devices and different identifiers are associated with them, RudderStack can easily track and merge them into the same virtual ID (`RUDDER_ID`) in the warehouse.
```
Comment (ishachopra0401) [inline identity-resolution.mdx:133]:
```suggestion
As the user logs in using more devices, different identifiers are associated with them. RudderStack easily tracks and merges them into the same virtual ID (`RUDDER_ID`) in the warehouse.
```

File: identity-resolution.mdx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        ]
+      }
+      mergeEvents.push(mergeEvent)
+    }
+  })
+  return events.concat(mergeEvents)
+}
+```
+
+### What happens in the warehouse?
```
Comment (ishachopra0401) [inline identity-resolution.mdx:119]:
```suggestion
#### What happens in the warehouse after merging the identifiers?
```

File: identity-resolution.mdx [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+Once you call the `merge` API or leverage the user transformations to merge the user identifiers and send the data to the warehouse destination, RudderStack creates the following two tables in your warehouse.
+
+The `RUDDER_IDENTITY_MAPPINGS` table contains the nodes (user identifiers and their values) associated with a given identity graph (customer profile). Each node has a  `RUDDER_ID` associated with it, as shown:
```
Comment (ishachopra0401) [inline identity-resolution.mdx:123]:
```suggestion
- `RUDDER_IDENTITY_MAPPINGS` table: It contains the nodes (user identifiers and their values) associated with a given identity graph (customer profile). Each node has a  `RUDDER_ID` associated with it, as shown:
```

File: identity-resolution.mdx [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+<img src="./assets/identity-resolution/identity-mappings-table.png" alt="Identity Mappings table in warehouse" />
+
+On the other hand, the `RUDDER_IDENTITY_MERGE_RULES` table contains all the edges that connect two different identifiers (nodes).
```
Comment (ishachopra0401) [inline identity-resolution.mdx:127]:
```suggestion
- `RUDDER_IDENTITY_MERGE_RULES` table: It contains all the edges that connect two different identifiers (nodes).
```

File: identity-resolution.mdx [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+
+The following three steps summarize the identity resolution process:
+
+1. Collect the user identifiers across various touchpoints (device IDs, email addresses, cookies, etc.)
+2. Connect these identifiers to an identity graph to build a unified user profile.
+3. Activate this profile by connecting it to various analytics, marketing, CRM tools as per the business requirement.
+
+## Identity resolution in RudderStack
+
+In modern businesses, the identity graph used for cross-device identity stitching can be quite large, sometimes consisting of hundreds of millions of nodes. However, as you gather more data, this number scales rapidly. In such a scenario, storing the identity graph in your data warehouse makes a lot of sense from a storage and scalability perspective.
```
Comment (ameypv-rudder) [inline identity-resolution.mdx:30]:
```suggestion
In modern businesses, the identity graph used for cross-device identity stitching can be quite large, sometimes consisting of hundreds of millions of nodes. As you gather more data, this number scales rapidly. In such a scenario, storing the identity graph in your data warehouse makes a lot of sense from a storage and scalability perspective.
```

File: identity-resolution.mdx [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+<div class="successBlock">
+
+With RudderStack's warehouse-first architecture, you can send all your cross-platform data to your warehouse and perform identity resolution on it.
+</div>
+
+This guide walks you through RudderStack's identity resolution feature in detail.
+
+## How identity resolution works
+
+Identity resolution involves using an **identity graph** - a database that houses and brings together all the different user identifiers and various aspects of their customer journeys into a single customer view. The identity graph collects and continually updates the customer profile with multiple identifiers mentioned above, like email, phone number, device IDs, etc.
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply identity-resolution.mdx:20]: Combined the existing sentence and your suggestion to say: "Identity resolution involves the usage of an identity graph...."

File: identity-resolution.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply identity-resolution.mdx:41]: Agreed- made the change.

File: identity-resolution.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply identity-resolution.mdx:50]: I agree - good call here. We don't even need to add the reference as the very next section talks about how the identifiers are merged.

File: identity-resolution.mdx (hunk 5 above)
Comment (ameypv-rudder) [reply identity-resolution.mdx:65]: Done.

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**